qml/pages/ChatPage.qml

```diff
@@ -681,1172 +681,1307 @@ Page {
-                onClicked: {
-                    if (chatPage.userIsMember) {
-                        var chatId = chatInformation.id;
-                        Remorse.popupAction(chatPage, qsTr("Leaving chat"), function() {
-                            tdLibWrapper.leaveChat(chatId);
-                            // this does not care about the response (ideally type "ok" without further reference) for now
-                            pageStack.pop(pageStack.find( function(page){ return(page._depth === 0)} ));
-                        });
-                    } else {
-                        tdLibWrapper.joinChat(chatInformation.id);
+        background: Column {
+            anchors.fill: parent
+            Row {
+                id: drawerHeaderRow
+                width: parent.width - ( 2 * Theme.horizontalPageMargin)
+                height: messageOptionsLabel.height + Theme.paddingLarge + ( chatPage.isPortrait ? ( 2 * Theme.paddingSmall ) : 0 )
+                anchors.horizontalCenter: parent.horizontalCenter
+                spacing: Theme.paddingMedium
+                Label {
+                    id: messageOptionsLabel
+                    text: qsTr("Additional Options")
+                    color: Theme.highlightColor
+                    font.pixelSize: Theme.fontSizeLarge
+                    width: parent.width - closeMessageOptionsButton.width - Theme.paddingMedium
+                    anchors.verticalCenter: parent.verticalCenter
+                    horizontalAlignment: Text.AlignRight
+
+                }
+                IconButton {
+                    id: closeMessageOptionsButton
+                    icon.source: "image://theme/icon-m-clear"
+                    anchors.verticalCenter: parent.verticalCenter
+                    onClicked: {
+                        messageOptionsDrawer.closeDrawer();
                     }
                 }
-                text: chatPage.userIsMember ? qsTr("Leave Chat") : qsTr("Join Chat")
             }
 
-            MenuItem {
-                id: muteChatMenuItem
-                visible: chatPage.userIsMember
-                onClicked: {
-                    var newNotificationSettings = chatInformation.notification_settings;
-                    if (newNotificationSettings.mute_for > 0) {
-                        newNotificationSettings.mute_for = 0;
-                    } else {
-                        newNotificationSettings.mute_for = 6666666;
+            Flickable {
+                id: drawerFlickable
+                width: parent.width
+                height: parent.height - drawerHeaderRow.height
+                contentHeight: drawerContentColumn.height
+                clip: true
+                Column {
+                    id: drawerContentColumn
+                    width: parent.width
+                    Repeater {
+                        model: messageOptionsDrawer.additionalItemsModel
+                        delegate: BackgroundItem {
```

Comment:
One could possibly put these BackgroundItems in a separate qml because they are basically the same thing, this may even reduce the parsing/compiling time a tiny bit. 
Also (you probably tried already): Doesn't a ContextMenu with MenuItems work here instead of the Column?

Reply:
> Also (you probably tried already): Doesn't a ContextMenu with MenuItems work here instead of the Column?

Or a SilicaListView with MenuItems in it.

Reply:
Yeah, that would probably make even more sense ;)

Reply:
Yesterday, I didn't make it to join two different lists of `NamedAction`s, but today it worked with probably the most embarrassing piece of JS code from the last century - but it seems to work.